### PR TITLE
add clear interval when using multiple inits

### PR DESCRIPTION
**Pull request by RobbeCl**
## Related Issues
Fixes https://github.com/nfriedly/express-rate-limit/issues/326

### Changed
Added check to memory store that when initialized multiple times, the previous interval is properly cleared. 

## Checklist

- [x] The issues that this PR fixes/closes have been mentioned above.
- [x] What this PR adds/changes/removes has been explained.
- [x] All tests (`npm test`) pass.
- [x] All added/modified code has been commented, and
      methods/classes/constants/types have been annotated with TSDoc comments.
- [x] If a new feature has been added or a bug has been fixed, tests have been
      added for the same.

### CODE REVIEW

File: test/library/memory-store-test.ts [hunk 1]
```diff
@@ -132,4 +132,16 @@ describe('memory store test', () => {
 			clearTimeout(realTimeoutId)
 		}
 	})
+
+	it('When calling init it should reset interval when already initialized', async () => {
```
Comment (gamemaker1) [inline test/library/memory-store-test.ts:136]: nit: could you please rename the test to: "should reset the interval incase it already exists"

File: test/library/memory-store-test.ts [hunk 2]
```diff
@@ -132,4 +132,16 @@ describe('memory store test', () => {
...
+		const shutdownSpy = jest.spyOn(MemoryStore.prototype, 'shutdown')
```
Comment (gamemaker1) [inline test/library/memory-store-test.ts:137]: Is the `.prototype` here necessary? If not, could you please drop it?

IIRC we should just need to pass the class to the `spyOn` function, no need to mess with prototype.

**Review by gamemaker1 — COMMENTED** [review]
Hi @RobbeCl,

Thanks for taking the time to make this PR! It looks good to me, just left two suggestions.

Also, is there any way we can make sure there are actually no more memory leaks? Maybe by adding another external test (tests in the `tests/external` folder, which are like mini projects in which to test the library's usage)?

**Comment by gamemaker1** [comment]
Also, it looks like the tests on Node 12 failed :/

**Comment by nfriedly** [comment]
I kinda feel like "you're holding it wrong", `init()` wasn't really meant to be called more than once. But, OK, we can add this in as a protection against that kind of abuse.

I'm not sure what's going on with the test failures - it appears that typescript isn't happy, but I'm not really sure why.

**Comment by RobbeCl** [comment]
@nfriedly So you must create a store per rate limiter then? But if I have like 20 rate limiters and want to close them, I need to call shutdown on each of them. Or is it weird to have 20 of them in the first place? :-)

**Comment by nfriedly** [comment]
@RobbeCl yeah, having 20 if them at the same time is a bit weird. For a testing scenario, I'd create a new instance (of the rate limiter and the store) for each test, and tear it down after the test. That way the behavior of one test isn't dependent on any other test.

**Pull request closed without merging**